### PR TITLE
s4742351 - 2D UNet

### PR DESCRIPTION
**Pull request by alex-strang**
Task: 2D UNet for prostate cancer diagnosis.
The pull request contains an implementation of a 2D UNet CNN used for assisting in prostate cancer diagnosis, via identification of points of interest. The code is program into several files, **dataset.py** which contains the data loaders for the relevant data, **modules.py** which contains the construction initial UNet model. **train.py** which contains the algorithm for training the model, and **predict.py** which shows an implementation of the trained model.

### CODE REVIEW

**Comment by yexincheng** [comment]
**This is an initial inspection, no action is required at this point**
1. **Solves problem**: The student could not access the training data (posted on Ed [#452](https://edstem.org/au/courses/18266/discussion/2340286)), thus, no training and prediction results were provided; the functionality of all scripts are not guaranteed.
2. **Implementation**: The model architecture (in the modules.py file) does not have a forward function, and it is a function rather than class. The train.py script does not specify batch size, learning rate, etc. The data passed into the network is not Tensor. No dataset class was implemented.
3. **Design**: The design is not good. No code demonstrates the dataset will be split into training and validation
4. **Comment**: In the module.py, the function returns "model", but in the comments, the returned argument is "NULL". No comments in dataset.py. Nice citations for other functions. In readme, dependency versions are not specified. It would be better to have reference for UNet paper and dataset.
5. **Difficulty**: **Easy**

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>No design and implementation. </td><td>-2</td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. </td><td>-4</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. 10 commits over 2 days.</td><td>-2</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. Incomplete.</td><td>-2</td></tr>
	<tr><td></td><td>Model/technical explanation :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). Submitted on time.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-23</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Incorrect folder structure -2, implementation incomplete -1 additionally

**Comment by shakes76** [comment]
Feedback applied +2
